g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductComponentAction.java

```diff
@@ -201,8 +201,14 @@ private String getMatchedVersion(final String productName, final String version)
 				}
 				for(String s : versionList) {
 					String f = filterVersion(s, version);
-					String ss = StringUtils.isEmpty(matchedVersion) ? s : filterVersion(s, matchedVersion);
-					if((compare(f, version) == -1 || (compare(f, version) == 0 && s.length() < version.length())) && compare(ss, matchedVersion) == 1) {
+					String ss = s;
+					String m = matchedVersion;
+					if(!StringUtils.isEmpty(matchedVersion)) {
```

Comment:
Move filterVersion into compare function, because here needs only compare result.
One suggestion: Assign the first version in the list to matchedVersion, then start comparison from second version.
This will make code and logic clearer.